### PR TITLE
Replace Jetbrains with Jakarta annotations

### PR DESCRIPTION
**Pull request by snazy**
`org.jetbrains.annotations.NotNull`/`Nullable` -> `jakarta.annotations.Nonnull`/`Nullable` plus Guava's `VisibleForTesting`

The Jakarta annotations are likely better supported by validation frameworks like Hibernate Validator.

### CODE REVIEW

**Comment by jbonofre** [comment]
NB: it's already done on Quarkus pr

**Comment by snazy** [comment]
> NB: it's already done on Quarkus pr

Correct. That's where it's from - to reduce the amount of changes from that PR

**Comment by jbonofre** [comment]
Love it 😀

**Comment by eric-maynard** [comment]
LGTM; but can we add this to the linter? I would hate to see us accidentally regress back to the wrong ones.

File: extension/persistence/eclipselink/build.gradle.kts [hunk 1]
```diff
@@ -45,7 +45,7 @@ dependencies {
     }
   }
 
-  compileOnly(libs.jetbrains.annotations)
+  compileOnly(libs.jakarta.annotation.api)
```
[the remark this answers is not in the document]
Comment (snazy) [reply extension/persistence/eclipselink/build.gradle.kts:48]: Yes, this

**Review by adutra — APPROVED** [review]

**Pull request merged**